testutils/src/main/scala/za/co/absa/enceladus/testutils/RandomDataGenerator.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.testutils
+
+import java.io.{BufferedWriter, File, FileWriter}
+
+import org.joda.time.DateTime
+import org.joda.time.format.{DateTimeFormat, DateTimeFormatter}
+
+import scala.util.Random
+
+object RandomDataGenerator {
+
+  // Set DateTime format to be used
+  val dtf: DateTimeFormatter = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss")
+
+  // Predefined columns
+  val reservedColumns: Map[Int,String] = Map(
+    1 -> "key",
+    2 -> "id",
+    3 -> "version",
+    4 -> "Date",
+    5 -> "value",
+    6 -> "address")
+
+  // Initialize random generator
+  private val random = new Random(System.nanoTime)
```

Comment:
This prevents reproducible results and complicates testing. Better would be to have the Random as an input parameter (with possible default) and make it the caller's responsibility to seed it.